src/cljs/org/broadinstitute/firecloud_ui/common/components.cljs

```diff
@@ -150,6 +162,47 @@
          content]])))})
 
 
+(react/defc Dialog
+  {:get-default-props
+   (fn []
+     {:blocking? true})
+   :render
+   (fn [{:keys [state props]}]
+     (when (:show-when props)
+       (let [content (:content props)]
+         (assert (react/valid-element? content)
+           (subs (str "Not a react element: " content) 0 200))
+         [:div {:style {:backgroundColor (if (:blocking? props)
+                                           "rgba(82, 129, 197, 0.4)"
+                                           "rgba(210, 210, 210, 0.4)")
+                        :overflowX "hidden" :overflowY "scroll"
+                        :position "fixed" :zIndex 9999
+                        :top 0 :right 0 :bottom 0 :left 0}
+                :onKeyDown (common/create-key-handler [:esc] #((:dismiss-self props)))
+                :onClick (when-not (:blocking? props) #((:dismiss-self props)))}
+          [:div {:style (if (:get-anchor props)
+                          {:position "absolute" :top (:anchor-top @state) :left (:anchor-left @state)}
+                          {:transform "translate(-50%, 0px)"
+                           :position "relative" :marginBottom 60
+                           :top 60 :left "50%" :width (:width props)})
+                 :onClick (when-not (:blocking? props) (fn [e] (.stopPropagation e)))}
+           content]])))
+   :component-did-mount
+   (fn [{:keys [this state props]}]
+     (when (:get-anchor props)
+       (let [domnode (.getDOMNode ((:get-anchor props)))
```

Comment:
Can you make this property `:get-anchor-dom-node` instead? That would make it safer for the next version of React.